arduinoRemote/app/src/main/java/com/example/arduinoremote/ConnectBT.java

```diff
@@ -3,32 +3,47 @@
 import android.bluetooth.BluetoothAdapter;
 import android.bluetooth.BluetoothDevice;
 import android.bluetooth.BluetoothSocket;
+import java.io.OutputStream;
 import android.os.AsyncTask;
+import android.util.Log;
+
 import java.io.IOException;
+import java.util.Set;
 import java.util.UUID;
 
 public class ConnectBT extends AsyncTask<Void, Void, Void> {
-    private boolean ConnectSuccess = true; //if it's here, it's almost connected
-    String address = "24:6F:28:B6:09:A2"; //this should be the arduino cars bt address
+    String carName = "Smartcar"; //this should be the arduino cars bt address
     BluetoothAdapter myBluetooth = null;
     BluetoothSocket btSocket = null;
-    private boolean isBtConnected = false;
-    static final UUID myUUID = UUID.fromString("25dd9b13-b489-4c34-a9b7-31fad9f63b4b");
+    BluetoothDevice device = null;
+    OutputStream btOutputStream;
 
 
     @Override
     protected Void doInBackground(Void... devices) { //while the progress dialog is shown, the connection is done in background
         try {
-            if (btSocket == null || !isBtConnected) {
+            Log.d("Testing","Background");
+
                 myBluetooth = BluetoothAdapter.getDefaultAdapter();//get the mobile bluetooth device
-                BluetoothDevice enhet = myBluetooth.getRemoteDevice(address);//connects to the device's address and checks if it's available
-                btSocket = enhet.createInsecureRfcommSocketToServiceRecord(myUUID);//create a RFCOMM (SPP) connection
-                BluetoothAdapter.getDefaultAdapter().cancelDiscovery();
-                btSocket.connect();//start connection
-            }
+
+                Set<BluetoothDevice> pairedDevices = myBluetooth.getBondedDevices();
+
+                if(pairedDevices.size() > 0) {
+
+                    for(BluetoothDevice device: pairedDevices) {
+
+                        if(device.getName().equals(carName)) {
+                            this.device = device;
+                            UUID uuid = UUID.fromString("00001101-0000-1000-8000-00805f9b34fb"); //Standard SerialPortService ID
```

Comment:
Should it be hard-coded?  --> "00001101-0000-1000-8000-00805f9b34fb"

Reply:
Yes